lib/active_merchant/billing/gateways/stripe_payment_intents.rb

```diff
@@ -423,17 +425,21 @@ def add_network_token_data(post_data, payment_method, options)
         post_data
       end
 
-      def add_network_token_cryptogram_and_eci(post, payment_method)
-        return unless adding_network_token_card_data?(payment_method)
+      def add_network_token_cryptogram_and_eci(post, payment_method, options)
+        return unless !options[:wallet_type] || adding_network_token_card_data?(payment_method, options)
```

Comment:
I think it would be better to update this to be something like 
`return unless payment_method.is_a?(NetworkTokenizationCreditCard) && options.dig(:stored_credential, :initiator) != 'merchant'`

Reply:
We only pass options[:wallet_type] when the payment method is a non tokenized GooglePay and in AM the payment method model for this would be CreditCard

Reply:
> I think it would be better to update this to be something like 
```
return unless payment_method.is_a(NetworkTokenizationCreditCard)
return if options[:stored_credential] && options.dig(:stored_credential, :initiator) == 'merchant'
```

Reply:
Yes, thanks!